src/graph/network/serialize.cc

```diff
@@ -0,0 +1,165 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file serialize.cc
+ * \brief Serialization for DGL distributed training.
+ */
+#include "serialize.h"
+#include "../network.h"
+
+#include <dmlc/logging.h>
+#include <dgl/immutable_graph.h>
+
+#include <cstring>
+
+namespace dgl {
+namespace network {
+
+const int kNumTensor = 7;  // We need to serialize 7 conponents (tensor) here
+
+int64_t SerializeSampledSubgraph(char* data,
+                                 const ImmutableGraph::CSR::Ptr csr,
+                                 const IdArray& node_mapping,
+                                 const IdArray& edge_mapping,
+                                 const IdArray& layer_offsets,
+                                 const IdArray& flow_offsets) {
```

Comment:
This interface makes changing the nodeflow structure subtle; one needs to also change the interface here (and the deserialization, and all the calling statements).

Is it possible to change this to something like `size_t SerializeNodeFlow(char *buffer, const NodeFlow &nf)`?

Reply:
No problem. I'll do this change after new NodeFlow stricture merged.

Reply:
If "new NodeFlow structure" refers to the one with optional tensor data (which is the one I'm using) then it won't be merged until after PR #453.  It's your call but I would prefer changing the interface now so later on NodeFlow could be extended further more easily.